algebird-core/src/main/scala/com/twitter/algebird/Ring.scala

```diff
@@ -144,14 +144,20 @@ object FloatRing extends Ring[Float] {
   override def minus(l: Float, r: Float): Float = l - r
   override def times(l: Float, r: Float): Float = l * r
 
+  // see: https://en.wikipedia.org/wiki/Kahan_summation_algorithm
+  // for this algorithm
   override def sumOption(t: TraversableOnce[Float]): Option[Float] =
     if (t.isEmpty) None
     else
       Some {
-        var sum = 0.0
         val iter = t.toIterator
+        var sum = iter.next().toDouble
```

Comment:
Any reason not to keep these as floats, and use `var c = 0.0f` for the aggregator?

Reply:
I made this change in #881, take it or leave it!